plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,277 @@
+# More Viable Plasma
+
+## Introduction
+
+[Minimal Viable Plasma](https://ethresear.ch/t/minimal-viable-plasma/426) (“Plasma MVP”) describes a simple specification for a UTXO-based Plasma chain. A key component of the Plasma MVP design is a protocol for “exits,” whereby a user may withdraw back to the root chain any funds available to them on the Plasma chain. The protocol presented in the specification requires users sign two signatures for every transaction. Concerns over the poor user experience presented by this protocol motivated the search for an alternative exit protocol. 
+
+In this document, we describe More Viable Plasma (“MoreVP”), a modification to the Plasma MVP exit protocol that improves user experience. The MoreVP exit protocol ensures the security of assets for clients correctly following the protocol. We initially present the most basic form of the MoreVP protocol and provide intuition towards its correctness.  We formalize certain requirements for Plasma MVP exit protocols and provide a proof that this protocol satisfies these requirements.
+
+An optimized version of the protocol is presented to account for restrictions of the Ethereum Virtual Machine. This optimization effectively caches the results of gas-heavy operations to decrease cost to the end user. A proof of the reduction from the optimized protocol to the basic protocol is provided to demonstrate that this optimized design also satisfies our formalized requirements. 
+
+We note the existence of certain attack vectors in the protocol, but find that most of these vectors can be largely mitigated and isolated to a relatively small attack surface. These attack vectors and their mitigations are described in detail. Although we conclude that the design is safe under certain reasonable assumptions about user behavior, some points are highlighted and earmarked for future consideration.
+
+Overall, the MoreVP exit protocol is a significant improvement over the original Plasma MVP exit protocol. We can further combine several optimizations to enhance user experience and reduce costs for users. Future work will focus on decreasing the implementation complexity of the design and continuing to minimize gas usage. 
+
+
+## Exit Protocol
+
+In this section, we specify the MoreVP exit protocol and give an intuitive argument toward its correctness. A formal treatment of the protocol is presented in the appendix. 
+
+
+### Definitions
+
+#### Deposit Transaction
+
+A deposit transaction is a special type of transaction that creates a new balance on the Plasma chain. This is similar to a Bitcoin "coinbase" transaction. In practice, this takes the form of a transaction with a single “special” input.
```

Comment:
I would add that for a deposit transaction to occur, user must deposit funds into the Plasma contract on rootchain.

Reply:
I leave this out because it's possible for the operator to create an invalid deposit transaction that does not correspond to a deposit.

Reply:
I would include both of those as potential deposit txns. 

This document should assume that people will read it as a guide and implement without doing 100% of the research that you've done - so i) deposit into plasma contract will ensure their understanding of how deposits interact with the rootchain and ii) operator invalid deposit will ensure they are aware of operator risks.